x-pack/plugins/ml/server/saved_objects/checks.ts

```diff
@@ -154,5 +156,106 @@ export function checksFactory(
     };
   }
 
-  return { checkStatus };
+  async function deleteJobsCheck(
+    request: KibanaRequest,
+    jobType: JobType,
+    jobIds: string[],
+    spacesEnabled: boolean,
+    resolveMlCapabilities: ResolveMlCapabilities
+  ) {
+    if (jobType !== 'anomaly-detector' && jobType !== 'data-frame-analytics') {
+      throw Boom.badRequest('Job type must be "anomaly-detector" or "data-frame-analytics"');
+    }
+
+    const mlCapabilities = await resolveMlCapabilities(request);
+    if (mlCapabilities === null) {
+      throw Boom.internal('mlCapabilities is not defined');
+    }
+
+    if (
+      mlCapabilities.canDeleteJob === false ||
+      mlCapabilities.canDeleteDataFrameAnalytics === false
+    ) {
+      // user does not have access to delete jobs.
+      return jobIds.reduce((results, jobId) => {
+        results[jobId] = {
+          canDelete: false,
+          canUnTag: false,
+        };
+        return results;
+      }, {} as DeleteJobCheckResponse);
+    }
+
+    if (spacesEnabled === false) {
+      // spaces are disabled, delete only no untagging
+      return jobIds.reduce((results, jobId) => {
+        results[jobId] = {
+          canDelete: true,
+          canUnTag: false,
+        };
+        return results;
+      }, {} as DeleteJobCheckResponse);
+    }
+    const canCreateGlobalJobs = await jobSavedObjectService.canCreateGlobalJobs(request);
+
+    const jobObjects = await Promise.all(
+      jobIds.map((id) => jobSavedObjectService.getJobObject(jobType, id))
+    );
+
+    return jobIds.reduce((results, jobId) => {
+      const jobObject = jobObjects.find((j) => j?.attributes.job_id === jobId);
+      if (jobObject === undefined || jobObject.namespaces === undefined) {
+        // job saved object not found
+        results[jobId] = {
+          canDelete: false,
+          canUnTag: false,
+        };
+        return results;
+      }
+
+      const { namespaces } = jobObject;
+      const isGlobalJob = namespaces.includes('*');
+
+      // job is in * space, user can see all spaces - delete and no option to untag
+      if (canCreateGlobalJobs && isGlobalJob) {
+        results[jobId] = {
+          canDelete: true,
+          canUnTag: false,
+          // GLOBAL JOB BOOL????
```

Comment:
Is this comment needed?

Reply:
it was a reminder that we might need information in this endpoint that the UI can use to inform the user why they can't untag or delete the job.
I'll remove the comment for now, but when the UI work is done we may need to add in flags like `isGlobalJob`

Reply:
fixed https://github.com/elastic/kibana/pull/83871/commits/ceb41ead974a82cb78fa916b5466015f2f654b78